cli/src/main/java/com/devonfw/tools/ide/merge/xml/XmlMerger.java

```diff
@@ -119,6 +124,17 @@ public Document merge(XmlMergeDocument templateDocument, XmlMergeDocument worksp
       if (strategy == null) {
         strategy = XmlMergeStrategy.COMBINE; // default strategy used as fallback
       }
+      if (this.legacyXmlSupport && (templateRoot.lookupNamespaceURI("merge") == null)) {
+        if (workspaceFileExists) {
+          strategy = XmlMergeStrategy.OVERRIDE;
+        } else {
+          strategy = XmlMergeStrategy.KEEP;
+        }
+      } else if (templateRoot.lookupNamespaceURI("merge") == null) {
+        this.context.warning(
+            "XML merge namespace not found. If you are working in a legacy devonfw-ide project, please set IDE_XML_MERGE_LEGACY_SUPPORT_ENABLED=true to "
+                + "proceed correctly.");
```

Comment:
Maybe there is another option to inform the user that the legacy environment variable has to be set to enable a correct merge behavior. Any suggestion?

Reply:
Totally fine for me. Legacy users can get spammed until they cleanup their stuff ;)